### PR TITLE
Feat/16290/add button component

### LINKED ISSUE #16290: [Ext Nav] Create component: Button
### Description
Create a reusable UI component for `Button` that unifies `ButtonPrimary`, `ButtonSecondary` and `ButtonLink`

### References
[Figma Link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=9970%3A48436)
[UI component guidelines](https://www.notion.so/UI-Components-5ccd2bf83eb8441892a0b72c0d8929e1)
[Styling guidelines](https://www.notion.so/Extension-Frontend-Engineering-Guide-b82ddb3e14004b5db3799a9b446294a9#49aa2ef7b7aa40178ff918043df71c34)
Testing guidelines: TBC

### Files needed
- `index.js`
- `index.scss`
- `button.js`
- `button.stories.js`
- `button.test.js`
- `README.mdx`

### Technical details
- [x] Uses a switch statement to render the different types of button: 
- [x] Accepts all props from the different buttons: ButtonPrimary, ButtonSecondary, ButtonLink

### Acceptance criteria
- [ ] Uses color, typography, shadows design tokens 
- [ ] Uses semantic html
- [ ] PropTypes have descriptions
- [ ] Has storybook story with controls
- [ ] Has documentation in MDX
- [ ] Has unit tests and 90% coverage
- [ ] Works in Chrome and Firefox
- [ ] Performance tested: no unnecessary re-renders or other performance concerns

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Unifying `ButtonPrimary`, `ButtonSecondary`, and `ButtonLink` to be `Button`. 

## More Information

Fixes: #16290 

### After
Screen Shot 2022-10-28 at 10 52 15 AM

## Manual Testing Steps
`yarn test:unit:jest .ui/components/component-library/button/button.test.js `

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [X] PR is linked to the appropriate GitHub issue
- [X] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [X] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/components/component-library/button/README.mdx [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+import { Button } from './button';
+
+### This is a base component. It should not be used in your feature code directly but as a "base" for other UI components
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/README.mdx:4]: Remove this
```suggestion
```

File: ui/components/component-library/button/README.mdx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+# Button
+
+The `Button` is the base component for buttons.
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/README.mdx:8]: Update the description

```suggestion
The `Button` is used for user actions it unifies `ButtonPrimary`, `ButtonSecondary` and `ButtonLink`
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good. Can we move all the ButtonBase updates into a separate PR for brevity?

File: ui/components/component-library/button-primary/button-primary.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ ButtonPrimary.propTypes = {
    */
   className: PropTypes.string,
   /**
-   * Boolean to change button type to Danger when true
+   * When true, ButtonSecondary color becomes Danger.
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-primary/button-primary.js:31]: Should be ButtonPrimary?
```suggestion
   * When true, `ButtonPrimary` color becomes Danger.
```

File: ui/components/component-library/button/README.mdx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+<Canvas>
+  <Story id="ui-components-component-library-button-button-stories-js--type" />
+</Canvas>
+
+```jsx
+import { Button, BUTTON_TYPES } from '../ui/component-library/button';
+
+<Button type={BUTTON_TYPES.PRIMARY}>Primary Button</Button>
+<Button type={BUTTON_TYPES.SECONDARY}>Secondary Button</Button>
+<Button type={BUTTON_TYPES.LINK}>Link Button</Button>
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/README.mdx:38]: To be consistent with out naming should we just use the names verbatim?
```suggestion
<Button type={BUTTON_TYPES.PRIMARY}>Button Primary</Button>
<Button type={BUTTON_TYPES.SECONDARY}>Button Secondary</Button>
<Button type={BUTTON_TYPES.LINK}>Button Link</Button>
```

File: ui/components/component-library/button/button.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * Select the type of Button.
+   * Possible values could be 'BUTTON_TYPES.PRIMARY', 'BUTTON_TYPES.SECONDARY', 'BUTTON_TYPES.LINK'
+   * Button will default to `BUTTON_TYPES.PRIMARY`
+   */
+  type: PropTypes.oneOf(Object.values(BUTTON_TYPES)),
+  /**
+   * Addition style properties to apply to the button.
+   */
+  style: PropTypes.object,
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/button.js:33]: Let's not promote the style prop in our components

```suggestion
```

File: ui/components/component-library/button/button.stories.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+      control: 'select',
+      options: Object.values(ICON_NAMES),
+    },
+    iconPositionRight: {
+      control: 'boolean',
+    },
+    iconProps: {
+      control: 'object',
+    },
+
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/button.stories.js:77]: nit: remove line break
```suggestion
```

File: ui/components/component-library/button/button.stories.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+export const Type = (args) => (
+  <Box display={DISPLAY.FLEX} gap={1}>
+    <Button type={BUTTON_TYPES.PRIMARY} {...args}>
+      Primary Button
+    </Button>
+    <Button type={BUTTON_TYPES.SECONDARY} {...args}>
+      Secondary Button
+    </Button>
+    <Button type={BUTTON_TYPES.LINK} {...args}>
+      Link Button
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/button.stories.js:128]: Let's update this text so it's consistent with our component naming
```suggestion
     Button Primary
    </Button>
    <Button type={BUTTON_TYPES.SECONDARY} {...args}>
      Button Secondary
    </Button>
    <Button type={BUTTON_TYPES.LINK} {...args}>
     Button Link
```

File: ui/components/component-library/button/README.mdx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+The `Button` is used for user actions it unifies `ButtonPrimary`, `ButtonSecondary` and `ButtonLink`
+
+<Canvas>
+  <Story id="ui-components-component-library-button-button-stories-js--default-story" />
+</Canvas>
+
+## Props
+
+The `Button` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/README.mdx:15]: Let's add that it accepts all prop for the other button types as well

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left some small suggestions

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Just need to fix linting. 🚢 it 💯

**Review by georgewrmarshall — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**